File: 파이썬/1.정렬./2750.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/2750
'''

# 버블 정렬
n = int(input())
l = []

for i in range(n) :
    m = int(input())
    l.append(m)

for i in range(n-1, -1, -1) :
    for j in range(0, i) :
        if l[j] > l[j+1] :
            tmp = l[j]
            l[j] = l[j+1]
            l[j+1] = tmp

for i in l :
    print(i, end = ' ')
    
             
# 선택 정렬
n = int(input())
l = []

for i in range(n) :
    m = int(input())
    l.append(m)

for i in range(n-1) :
    key = i
    for j in range(i+1, n) :
         if l[key] > l[j] :  
             key = j 
    tmp = l[key]
    l[key] = l[i]
    l[i] = tmp
         
for i in l :
    print(i, end = ' ')
    
# 삽입 정렬
n = int(input())
l = []

for i in range(n) :
    m = int(input())
    l.append(m)

for i in range(1, n) :
    key = l[i] 
    j = i-1 
    while j>=0 and l[j] > key :
        l[j+1] = l[j]
        j -=1 
    l[j+1] = key
    
for i in l :
    print(i, end = ' ')    

File: C++/정렬알고리즘 - 초급/2750.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/2750
*/

// 버블 정렬
#include <iostream>

using namespace std;

int arr[1000];
int n;

int main(){

    cin >> n;
    for(int i=0; i<n; i++){
        cin >> arr[i];
    }

    for(int i=n-1; i>=0; i--){
        for(int j=0; j<i; j++){
            if(arr[j] > arr[j+1]){
                int tmp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = tmp;
            }
        }
    }

    for(int i=0; i<n; i++){
        cout << arr[i] << "\n";
    }
    
    return 0;
}

// 선택 정렬
#include <iostream>

using namespace std;

int arr[1000];
int n;

int main(){

    cin >> n;
    for(int i=0; i<n; i++){
        cin >> arr[i];
    }

    for(int i=0; i<n-1; i++){
        int key = i;
        for(int j=i+1; j<n; j++){
            if(arr[key] > arr[j]){
                key = j;
            }
        }
        int tmp = arr[key];
        arr[key] = arr[i];
        arr[i] = tmp;
    }

    for(int i=0; i<n; i++){
        cout << arr[i] << "\n";
    }
    
    return 0;
}

// 삽입 정렬
#include <iostream>

using namespace std;

int arr[1000];
int n, j;

int main(){

    cin >> n;
    for(int i=0; i<n; i++){
        cin >> arr[i];
    }

    for(int i=1; i<n; i++){ 
        int key = arr[i]; 
        for(j=i-1; j>=0 && arr[j] > key; j--){
            arr[j+1] = arr[j]; 
        } 
        arr[j+1] = key; 
    }

    for(int i=0; i<n; i++){
        cout << arr[i] << "\n";
    }
    
    return 0;
}